awscli/customizations/autoprompt/history.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+from prompt_toolkit.completion import Completion, Completer
+from prompt_toolkit.history import FileHistory
+
+from awscli.autocomplete.completer import CompletionResult
+from awscli.autocomplete.filters import fuzzy_filter
+
+
+class HistoryDriver(FileHistory):
+    HISTORY_VERSION = 1
+
+    def __init__(self, filename, max_commands=500):
+        # keep only last max_commands commands in file, each command
+        # takes three lines
+        super().__init__(filename)
+        self._max_commands = max_commands
+
+    def load_history_strings(self):
+        try:
+            with open(self.filename, 'r') as f:
+                commands = json.load(f).get('commands', [])
+            return reversed(commands)
+        except Exception:
+            return []
+
+    def store_string(self, string):
+        history = {'version': self.HISTORY_VERSION, 'commands': []}
+        try:
+            if os.path.exists(self.filename):
+                with open(self.filename, 'r') as f:
+                    history = json.load(f)
+            history['commands'].append(string)
+            history['commands'] = history['commands'][-self._max_commands:]
+            with open(self.filename, 'w') as f:
+                json.dump(history, f)
+        except Exception:
+            pass
```

Comment:
So one thing that I think that would be valuable here is to log what the exception was here in the case we are trying to trouble shoot why history is not showing up from a bug report. However, I also noticed that we do not enable debug logs before we enter in auto-prompt mode so for this case, it would not output the message if there was an issue.

I think for this PR we should instantaite a logger similar to what we do in other modules: https://github.com/aws/aws-cli/blob/63c1e9182c3411a1b1b02baff80b33bc9ecdd8de/awscli/customizations/commands.py#L19 and log the exception. Then in a follow up PR we need to figure out the story around debugging this because this will be important before we release this feature.